lib/providers/update_provider.dart

```
@@ -0,0 +1,372 @@
+    required this.description,
+    required this.publishedAt,
+  });
+
+  @override
+  String toString() =>
+      'UpdateVersion(version: $version, description: $description, publishedAt: $publishedAt)';
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+
+    return other is UpdateVersion &&
+        other.version == version &&
+        other.description == description &&
+        other.publishedAt == publishedAt;
+  }
+
+  @override
+  int get hashCode =>
+      version.hashCode ^ description.hashCode ^ publishedAt.hashCode;
+
+  static const titleField = 'title';
+  static const descriptionField = 'description';
+  static const publishedAtField = 'pubDate';
+
+  static const rpm = 'rpm';
+  static const deb = 'deb';
+  static const tarball = 'tar.gz';
+  static const appImage = 'appimage';
+  static const linuxDownloadFileName = 'bluecherry-linux-x86_64';
+  static const windowsDownloadFileName = 'bluecherry-windows-setup';
+}
+
+class UpdateManager extends ChangeNotifier {
+  UpdateManager._();
+
+  /// `late` initialized [UpdateManager] instance.
+  static late final UpdateManager instance;
+  late final PackageInfo packageInfo;
+  late String tempDir;
+
+  /// The URL to the appcast file.
+  static const appCastUrl =
+      'https://raw.githubusercontent.com/bdlukaa/unity/upgrade/bluecherry_appcast.xml';
+
+  /// Initializes the [UpdateManager] instance & fetches state from `async`
+  /// `package:hive` method-calls. Called before [runApp].
+  static Future<UpdateManager> ensureInitialized() async {
+    instance = UpdateManager._();
+    await instance.initialize();
+    return instance;
+  }
+
+  /// If true, there is a new update available.
+  ///
+  /// If false, the user is up to date with the latest version.
+  bool get hasUpdateAvailable {
+    final currentVersion = packageInfo.version;
+    return Version.parse(currentVersion) !=
+        Version.parse(latestVersion.version);
```

It is necessary to do a check if the latest version is not lower than the current version. It should throw an error